Ft create employee 171513935

katiamiyu opened the pull request:
### What does this PR do?
create a new employee record
view all existing records
edit existing records
remove existing records
run test on each endpoint
#### Description of Task to be completed?
- manage employee endpoints
- create a test response value
- create reasonable and user-friendly responses
#### How should this be manually tested?
- clone the repository
- cd into the repository
- use "npm i" to download all dependencies
- add .env file to the root folder
- run npm run pretest
- run npm test
#### Any background context you want to provide?
- get the test database connection string
- contact me "katiamiyu@gmail.com"
#### What are the relevant pivotal tracker stories?
- https://www.pivotaltracker.com/n/projects/2436135
- user can perform functions on employee

katiamiyu committed:
feat(create employee):

enter employee detail
check employee entry values
create new employee

[Finishes #171513935]

katiamiyu committed:
feat(create employee):

enter employee detail
check employee entry values
create new employee

[Finishes #171513935]

katiamiyu committed:
feat(get all employee):

enter valid endpoint address
check employees available in database
retrieve all available employees

[Finishes #171513959]

katiamiyu committed:
feat(get employee by id):

enter valid endpoint address
check employees available in database
retrieve all available employees

[Finishes #171514089]

katiamiyu committed:
feat(remove employee by id):

enter valid endpoint address
check employees available in database
remove  available employees with id

[Finishes #171513837]

JamesUgbanu reviewed (changes requested):
I found three(3) issues with your code
1. I stripped checkid into an helper function, so we don't keep repeating ourselves
2. You are running another query inside update when you have a function that already takes care of that - see line 59
3. validation middleware does not need conn included because we are not querying the database

katiamiyu committed:
feat(remove employee by id):

enter valid endpoint address
check employees available in database
remove  available employees with id

[Finishes #171513837]

katiamiyu committed:
Merge branch 'develop' into ft-create-employee-171513935

katiamiyu committed:
feat(remove employee by id):

enter valid endpoint address
check employees available in database
remove  available employees with id

[Finishes #171513837]

katiamiyu commented:
> I found three(3) issues with your code
> 
> 1. I stripped checkid into an helper function, so we don't keep repeating ourselves
> 2. You are running another query inside update when you have a function that already takes care of that - see line 59
> 3. validation middleware does not need conn included because we are not querying the database


Thanks for the review. codes have been updated now.
